File: languages/go/entity/identityV2/credential_schema.go
package identityV2

import "github.com/bloock/bloock-sdk-go/v2/internal/bridge/proto"

// CredentialSchema represents the schema information for a credential, including its ID and type.
type CredentialSchema struct {
	Id   string
	Type string
}

func NewCredentialSchemaFromProto(s *proto.CredentialSchemaV2) CredentialSchema {
	if s == nil {
		return CredentialSchema{}
	}
	return CredentialSchema{
		Id:   s.Id,
		Type: s.Type,
	}
}

func (c CredentialSchema) ToProto() *proto.CredentialSchemaV2 {
	return &proto.CredentialSchemaV2{
		Id:   c.Id,
		Type: c.Type,
	}
}

File: languages/python/bloock/entity/identity_v2/credential_schema.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_v2_pb2 as proto


class CredentialSchema:
    """
    Represents the schema information for a credential, including its ID and type.
    """
    def __init__(self, id: str, type: str) -> None:
        """
        Constructs an CredentialSchema object with the specified parameters.
        :type type: object
        :type id: object
        :rtype: object
        """
        self.id = id
        self.type = type

    @staticmethod
    def from_proto(c: proto.CredentialSchemaV2) -> CredentialSchema:
        return CredentialSchema(
            id=c.id,
            type=c.type
        )

    def to_proto(self) -> proto.CredentialSchemaV2:
        return proto.CredentialSchemaV2(
            id=self.id,
            type=self.type
        )
